tests/tensor_search/integ_tests/test_search_structured.py

```diff
@@ -1184,3 +1186,115 @@ def test_tensor_search_with_custom_vector_query(self):
+
+    def test_special_characters_in_map_score_modifiers(self):
+         special_characters = [
+             # Basic punctuation
+             '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '-', '+', '=', '{', '}', '[', ']', ':', ';', "'", '"',
+             '<', '>', ',', '.', '?', '/', '\\', '|', '~', '`',
+
+             # Math symbols
+             '+', '-', '*', '/', '=', '%', '<', '>', '±', '≠', '≈', '√', '∞', '∑', '∫', '∂', '∇', '∏', '⊕', '∪', '∩',
+
+             # Currency symbols
+             '$', '€', '£', '¥', '₹', '₽', '₿', '¢', '₩', '₦', '₫',
+
+             # Accented and diacritic letters
+             'á', 'é', 'í', 'ó', 'ú', 'ü', 'ñ', 'ç', 'ø', 'å', 'œ', 'æ', 'ÿ', 'ä', 'ö', 'ß', 'ø',
+
+             # Greek letters
+             'α', 'β', 'γ', 'δ', 'ε', 'ζ', 'η', 'θ', 'λ', 'μ', 'π', 'φ', 'ψ', 'ω',
+
+             # Arrows and other symbols
+             '←', '↑', '→', '↓', '↔', '↕', '⇐', '⇒', '⇔', '∞', '∴', '≈', '≠', '≥', '≤', '√',
+
+             # Box drawing characters
+             '─', '│', '┌', '┐', '└', '┘', '┼', '╳', '█', '░', '▒', '▓',
+
+             # Emoji
+             '🙂', '😁', '😂', '🤔', '🙄', '😎', '😢', '😡', '😍', '🤯', '🐱', '🐶', '🌟', '🌍', '🚀', '🍕', '🎉', '❤️',
+
+             # Superscripts and subscripts
+             '¹', '²', '³', '⁴', '⁵', '₁', '₂', '₃', '₄', '₅',
+
+             # Other symbols
+             '☂', '⚡', '☯', '✈', '✉', '⚽', '♠', '♣', '♥', '♦', '☢', '⚠', '☮',
+         ]
+
+         failed_characters = []
+         soft_failed_characters = []
+         supported_characters = []
+
+         for special_character in special_characters:
+             try:
+                 docs = [
+                     {
+                         "_id": "1_map",
+                         "text_field_1": "a photo of a cat",
+                         "map_score_mods_float": {f"a{special_character}subsubfield": 0.5},
+                     }
+                 ]
+
+                 add_result = self.add_documents(
+                     config=self.config,
+                     add_docs_params=AddDocsParams(
+                         index_name=self.default_text_index,
+                         docs=docs,
+                     )
+                 )
+
+                 score_modifiers = ScoreModifierLists(**{
+                     "add_to_score": [{"field_name": f"map_score_mods_float.a{special_character}subsubfield", "weight": 2}],
+                 })
+
+                 res = tensor_search.search(
+                     config=self.config,
+                     index_name=self.default_text_index,
+                     text="",
+                     score_modifiers=score_modifiers,
+                 )
+
+                 expected_score = 1.4165449318484857
+                 actual_score = res['hits'][0]['_score']
+
+                 if abs(actual_score - expected_score) > 0.01:
+                     #print(f"Soft fail for special character '{special_character}': Score mismatch")
```

Comment:
can we remove the commented out code from tests?

Reply:
I wanted to leave them commented out for future debugging. What do you think?

Reply:
Removed!